src/load/azext_load/data_plane/load_test/custom.py

```diff
@@ -302,6 +307,93 @@ def convert_to_jmx(
     return response.as_dict()
 
 
+def set_baseline(
+    cmd,
+    load_test_resource,
+    test_id,
+    test_run_id,
+    resource_group_name=None,
+):
+    logger.info("Setting baseline for test with test ID: %s", test_id)
+    test_client = get_admin_data_plane_client(cmd, load_test_resource, resource_group_name)
+    test_run_client = get_testrun_data_plane_client(cmd, load_test_resource, resource_group_name)
+    existing_test = test_client.get_test(test_id)
+    existing_test_run = test_run_client.get_test_run(test_run_id)
+    if existing_test_run.get("testId") != test_id:
+        raise InvalidArgumentValueError(
+            f"Test run with ID: {test_run_id} is not associated with test ID: {test_id}"
+        )
+    if existing_test_run.get("status") not in ["CANCELLED", "DONE"]:
+        raise InvalidArgumentValueError(
+            f"Test run with ID: {test_run_id} does not have a valid "
+            f"test run status {existing_test_run.get('status')}. "
+            "Valid test run status are: CANCELLED, DONE"
+        )
+    if existing_test_run.get("testRunStatistics", {}).get("Total") is None:
```

Comment:
Since portal doesn't allow setting of baseline until statistics are available, had discussed with Saloni and was asked to keep similar as portal client. Also, it seemed like a good idea to wait as if the compare-to-baseline is immediately executed, baseline test run might be missed